stable_baselines/common/segment_tree.py

```diff
@@ -61,29 +62,42 @@ def reduce(self, start=0, end=None):
         return self._reduce_helper(start, end, 1, 0, self._capacity - 1)
 
     def __setitem__(self, idx, val):
-        # index of the leaf
-        idx += self._capacity
-        self._value[idx] = val
-        idx //= 2
-        while idx >= 1:
-            self._value[idx] = self._operation(
-                self._value[2 * idx],
-                self._value[2 * idx + 1]
-            )
-            idx //= 2
+        # indexes of the leaf
+        idxs = idx + self._capacity
+        self._value[idxs] = val
+        if isinstance(idxs, np.ndarray):
```

Comment:
Any specific reason to also support non-ndarray versions? I think `SegmentTree` can also be updated to use `ndarray`s by default (for `self._value`), so any inheriting classes do not have to manually change it.

Reply:
Since the vectorized version completely rebuilds the whole tree, the version in the else statement should be more efficient for updating single items. If that is not done often, I could unify it.

Reply:
Since `np.any(idxs >= 1)` works for scalars too (non-ndarray), you can remove the other block (the one for updating single item). I do not expect it to affect the performance significantly and readability of the code outweighs this.